scripts/promote-catalog.py

```diff
@@ -0,0 +1,486 @@
+
+    Args:
+        operator_name (str): Operator name
+        repo_dir (str): Path to the processed local repository
+        source (str): Source catalog directory
+        target (str): Target catalog directory
+
+    Returns:
+        list[str]: List of relative paths to copied files
+        for Git commit
+    """
+    to_commit = []
+
+    # copy catalog files if they don't exist
+    relative_catalog_dir = os.path.join("catalogs", f"v{target_version}", operator_name)
+    catalog_target_dir = os.path.join(repo_dir, relative_catalog_dir)
+    if not os.path.exists(catalog_target_dir):
+        os.makedirs(catalog_target_dir)
+    catalog_source_dir = os.path.join(
+        repo_dir, "catalogs", f"v{source_version}", operator_name
+    )
+    for file in os.listdir(catalog_source_dir):
+        source_file = os.path.join(catalog_source_dir, file)
+        target_file = os.path.join(catalog_target_dir, file)
+        if not os.path.exists(target_file):
+            LOG.info("Copying catalog %s to %s", source_file, target_file)
+            shutil.copyfile(source_file, target_file)
+            to_commit.append(os.path.join(relative_catalog_dir, file))
+
+    # copy template file if they don't exist
+    relative_template_file = os.path.join(
+        "operators", operator_name, "catalog-templates", f"v{target_version}.yaml"
+    )
+    source_template = os.path.join(
+        repo_dir, f"operators/{operator_name}/catalog-templates/v{source_version}.yaml"
+    )
+    target_template = os.path.join(repo_dir, relative_template_file)
+    if not os.path.exists(target_template):
+        LOG.info("Copying template %s to %s", source_template, target_template)
+        shutil.copyfile(source_template, target_template)
+        to_commit.append(relative_template_file)
+
+    return to_commit
+
+
+def update_makefile(
+    operator_name: str,
+    repo_dir: str,
+    target_version: str,
+) -> str:
+    """
+    Add target version to the `OCP_VERSIONS` in the Makefile if missing
+
+    Args:
+        operator_name (str): Operator name
+        repo_dir (str): Path to the processed local repository
+        target_version (str): Target version to add
+    """
+    to_commit = ""
+
+    relative_makefile_path = os.path.join("operators", operator_name, "Makefile")
+    makefile_path = os.path.join(repo_dir, relative_makefile_path)
+    with open(makefile_path, "r") as makefile:
+        makefile_content = makefile.readlines()
+
+    makefile_out = []
+    for line in makefile_content:
+        if line.startswith("OCP_VERSIONS"):
+            if target_version not in line:
+                # assuming line looks like:
+                # `OCP_VERSIONS=$(shell echo "v4.12 v4.13 v4.14 v4.15" )`
+                split_point = line.rindex('"')
+                updated = f"{line[:split_point]} v{target_version}{line[split_point:]}"
+                makefile_out.append(updated)
+                to_commit = relative_makefile_path
```

Comment:
I did a little refactor here: I kept the `starts_with` but added the additional `=` to identify variable assignment. Used regex for the versions (with both single and double quotes) and do the `not in` check above list to avoid false positives. Added a comment to the template Makefile to make it clearer this will be handled by automation.

Reply:
Changed to regex as per response in other comment here.